lib/application_insights/channel/contracts/data.rb

```diff
@@ -1,48 +1,9 @@
 require_relative 'json_serializable'
 
-module ApplicationInsights
-  module Channel
-    module Contracts
-      # Data contract class for type Data.
-      class Data < JsonSerializable
-        # Initializes a new instance of the Data class.
-        def initialize(options={})
-          defaults = {
-            'baseType' => nil,
-            'baseData' => nil
-          }
-          values = {
-            'baseData' => nil
-          }
-          super defaults, values, options
-        end
-        
-        # Gets the base_type property.
-        def base_type
-          @values.fetch('baseType') { 
-            @values['baseType'] = nil
-          }
-        end
-        
-        # Sets the base_type property.
-        def base_type=(value)
-          if value == @defaults['baseType']
-            @values.delete 'baseType' if @values.key? 'baseType'
-          else
-            @values['baseType'] = value
-          end
-        end
-        
-        # Gets the base_data property.
-        def base_data
-          @values['baseData']
-        end
-        
-        # Sets the base_data property.
-        def base_data=(value)
-          @values['baseData'] = value
-        end
-      end
-    end
+module ApplicationInsights::Channel::Contracts
+  class Data
+    include JsonSerializable
+
+    attributes :baseType, :baseData
```

Comment:
I like your proposal to keep the code with Ruby's style and let the _presenter_ uses the mappings instead making a transformation.

I have a few examples in my mind that could fit with your idea:

## Example 1
Accessors and serialization field names are defined in just one method call.
To define default values we need to override the _get attribute_.

```ruby
module ApplicationInsights::Channel::Contracts
  class User
    include JsonSerializable

    attributes account_acquisition_date: 'ai.user.accountAcquisitionDate',
               account_id: 'ai.user.accountId',
               user_agent: 'ai.user.userAgent',
               id: 'ai.user.id'  

  end
end
```

## Example 2
Same as **Example 1** but allows to set the defaults in the same hash.
We can consider a different syntax, [like an ORM does](https://github.com/emancu/ork#attribute)

```ruby
module ApplicationInsights::Channel::Contracts
  class User
    include JsonSerializable

    attributes account_acquisition_date: {name: 'ai.user.accountAcquisitionDate', default: Time.now},
               account_id: {name: 'ai.user.accountId'},
               user_agent: {name: 'ai.user.userAgent'},
               id: {name: 'ai.user.id', default: 1}

  end
end
```

## Example 3
Given most of the contracts uses the same pattern between other contracts and also with inner attributes we can create Decorators in order to transform the values and keep the simplicity.

```ruby
module ApplicationInsights::Channel::Contracts
  class User
    include JsonSerializable

    attributes :account_acquisition_date, :account_id, :user_agent, :id
    decorate_on_serialization_with CamelCaseDecorator.new
    decorate_on_serialization_with PrefixDecorator.new('ai.user.')

  end
end
```

We will specify several decorators which will be applied in order and also define custom decorators which receives a block and transform the value as the user wants.

```ruby
CustomDecorator.new do |name|
  "ai.user.#{name.to_camel_case}"
end
```

Which one fits better for you?